nemo_skills/evaluation/evaluator/mmau_pro.py

```diff
@@ -0,0 +1,354 @@
+
+
+def evaluate_with_nvembed_similarity(
+    model_prediction: str, choices: list, ground_truth: str, config: MMAUProEvaluatorConfig
+) -> tuple[str, float]:
+    """NVEmbed-based evaluation: match predictions to choices using embedding similarity."""
+    import torch
+    import torch.nn.functional as F
+
+    model = load_nvembed_model(config.embedding_model)
+    device = next(model.parameters()).device
+
+    with torch.no_grad():
+        prediction_embedding = model.encode([model_prediction], instruction="", max_length=4096, device=device)
+        choice_embeddings = model.encode(choices, instruction="", max_length=4096, device=device)
+
+    prediction_embedding = F.normalize(prediction_embedding, p=2, dim=1)
+    choice_embeddings = F.normalize(choice_embeddings, p=2, dim=1)
+
+    scores = (prediction_embedding @ choice_embeddings.T) * 100
+    scores = scores.squeeze()
+
+    if scores.dim() == 0:
+        scores = scores.unsqueeze(0)
+
+    best_choice_idx = torch.argmax(scores).item()
+    matched_choice = choices[best_choice_idx]
+    confidence = torch.max(scores).item()
+
+    return matched_choice, confidence
+
+
+def load_nvembed_model(model_name: str = "nvidia/NV-Embed-v2"):
+    """Load NVEmbed model using HuggingFace AutoModel with GPU support."""
+
+    import os
+
+    import torch
+    from transformers import AutoModel
+
+    if not hasattr(load_nvembed_model, "_cache"):
+        load_nvembed_model._cache = {}
+
+    if model_name in load_nvembed_model._cache:
+        return load_nvembed_model._cache[model_name]
+
+    device = "cuda" if torch.cuda.is_available() else "cpu"
+
+    # Use explicit cache directory (respects HF_HOME env var)
+    cache_dir = os.environ.get("HF_HOME")
+    if cache_dir:
+        LOG.info(f"Using HuggingFace cache directory: {cache_dir}")
+
+    try:
+        model = AutoModel.from_pretrained(
+            model_name, trust_remote_code=True, cache_dir=cache_dir, local_files_only=False
+        )
+        model.to(device)
+        model.eval()
+        load_nvembed_model._cache[model_name] = model
+        LOG.info(f"Successfully loaded {model_name} on {device}")
+        return model
+    except Exception as e:
+        LOG.error(f"Failed to load {model_name}: {e}")
+        raise
+
+
+def evaluate_closed_form_with_nvembed(sample: dict[str, Any], config: MMAUProEvaluatorConfig) -> dict[str, Any]:
+    """Evaluate closed-form questions using NVEmbed similarity matching."""
+
+    if "nvembed_confidence" in sample:
+        LOG.info("Skipping sample - nvembed_confidence already exists")
+        return sample
+
+    generation = sample.get("generation", "").strip()
+    choices = sample.get("choices", [])
+    expected_answer = sample.get("expected_answer", "")
+
+    LOG.info(f"NVEmbed evaluation for generation='{generation[:50]}'")
+
+    if not generation or not choices:
+        LOG.warning("Missing generation or choices for nvembed evaluation")
```

Comment:
can this normally happen? If not, we should probably just fail here and let users investigate why the input is broken

Reply:
same applies to other functions inside this file. It's best to not hide any potential issues, if something isn't as expected, just let the whole job fail